### PR TITLE
PDS-11 Login Logout Functionality

### PR DESCRIPTION
**Pull request by m-bilal-asghar**
### Description:
- Adds login functionality
- Adds log out functionality.
- Adds navigation hook.
- Created home component.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -5,9 +5,11 @@ import { collection, getDocs } from "firebase/firestore";
 import { db } from "./firebase-config";
 import Navbar from "./components/Navbar";
 import Login from "./components/Login";
+import LogOut from "./components/Login/LogOut";
```
Comment (usamamushtaqdevsinc) [inline src/App.js]: why O is capital in `LogOut`? and why not `Login` like LogIn?  Try to use same patterns

File: src/components/Login/LogOut.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { useNavigate } from "react-router-dom";
+import { signOut } from "firebase/auth";
+
+import { auth } from "../../firebase-config";
+
+const LogOut = () => {
+  const nav = useNavigate();
+
+  useEffect(() => {
+    const logOut = async () => {
```
Comment (usamamushtaqdevsinc) [inline src/components/Login/LogOut.jsx]: move this method to outside the useEffect

File: src/components/Login/index.jsx [hunk 3]
```diff
@@ -1,26 +1,39 @@
...
+  const [ loginData, setLoginData ] = useState({
+    email: "",
+    password: "",
+  });
+
+  const nav = useNavigate();
 
   const changeHandlerData = (event) => {
     const { name, value } = event.target;
     setData({ ...data, [name]: value });
```
Comment (usamamushtaqdevsinc) [inline src/components/Login/index.jsx:29]: can we try this?
`{...data, name: value }`

File: src/components/Login/index.jsx [hunk 4]
```diff
@@ -48,6 +62,18 @@ const Login = () => {
     });
   }
 
+  const loginAuthUser = async (event) => {
+    event.preventDefault()
+    await signInWithEmailAndPassword(
```
Comment (usamamushtaqdevsinc) [inline src/components/Login/index.jsx:67]: move into a separate helper file

File: src/App.js (hunk 1 above)
Comment (m-bilal-asghar) [reply src/App.js]: OK I'll will follow Login/Logout convention.

File: src/components/Login/LogOut.jsx (hunk 2 above)
Comment (m-bilal-asghar) [reply src/components/Login/LogOut.jsx]: Sure :+1:

File: src/components/Login/index.jsx (hunk 3 above)
Comment (m-bilal-asghar) [reply src/components/Login/index.jsx:29]: No. I don't think so. Because [name] represent dynamic keys within hash.

**Pull request merged**